server/matchmaker/algorithm.py

```diff
@@ -41,64 +30,45 @@ def _unmatch(self, s1: Search):
 
 
 class StableMarriage(MatchmakingPolicy):
-    def find(self) -> Dict[Search, Search]:
-        """Perform SM_NUM_TO_RANK runs of the stable matching algorithm. 
-        Assumes that _MatchingGraph.build_sparse() only returns edges whose matches are acceptable
-        to both parties."""
-        ranks = _MatchingGraph.build_sparse(self.searches)
+    def find(self, ranks: WeightedGraph) -> Dict[Search, Search]:
+        """ Perform the stable matching algorithm until a maximal stable matching
+        is found.
+        """
         self.matches.clear()
 
-        for i in range(SM_NUM_TO_RANK):
+        max_degree = max((len(edges) for edges in ranks.values()), default=0)
+        for i in range(max_degree):
```

Comment:
If variable i is not used, in python we use underscore.

Reply:
It is used for logging